packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -578,7 +583,14 @@ export default class CodeGen {
         // Depth-first traversal. We assign a partId to each element, which is an integer based on traversal order.
         while (stack.length > 0) {
             const node = stack.shift()!;
-            partId++;
+
+            if (!(isComment(node) && !this.preserveComments)) {
```

Comment:
This is easier to read :)
```suggestion
            // Skip comment nodes in parts count, as they will be stripped in production, unless when `preseveComments` is enabled
            if (!isComment(node) || this.preserveComments)) {
```